cloud/metadata/github.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+//nolint:revive
+package metadata
+
+import (
+	"time"
+
+	"github.com/google/go-github/v58/github"
+	"github.com/terramate-io/terramate/strconv"
+)
+
+type GithubCommit struct {
+	// GithubAuthor is the Github author from the Github perspective.
+	GithubAuthor *GithubUser `json:"github_author,omitempty"`
+
+	// GitAuthor is the Git author from the Github perspective.
+	GitAuthor *CommitAuthor `json:"git_author,omitempty"`
+
+	// GithubCommitter is the Github committer from the Git perspective.
+	GithubCommitter *GithubUser `json:"github_committer,omitempty"`
+
+	// GitCommitter is the Git committer from the Git perspective.
+	GitCommitter *CommitAuthor `json:"git_committer,omitempty"`
+}
+
+type GithubPullRequest struct {
+	// Author of the PR.
+	User *GithubUser `json:"user,omitempty"`
+	// Author of latest commit on the PR branch.
+	HeadUser *GithubUser `json:"head_user,omitempty"`
+	// Author of latest commit on base branch.
+	BaseUser *GithubUser `json:"base_user,omitempty"`
+
+	RequestedReviewers []GithubUser `json:"requested_reviewers,omitempty"`
+	RequestedTeams     []GithubTeam `json:"requested_teams,omitempty"`
+
+	Reviews []GithubPullRequestReview `json:"reviews,omitempty"`
+
+	Assignees []GithubUser `json:"assignees,omitempty"`
+}
+
+type GithubUser struct {
+	ID         string `json:"id,omitempty"`
+	NodeID     string `json:"node_id,omitempty"`
+	Login      string `json:"author_login,omitempty"`
+	Name       string `json:"author_gravatar_id,omitempty"`
+	AvatarURL  string `json:"avatar_url,omitempty"`
+	GravatarID string `json:"gravatar_id,omitempty"`
+	Email      string `json:"email,omitempty"`
+}
+
+type CommitAuthor struct {
+	Name  string     `json:"name,omitempty"`
+	Email string     `json:"email,omitempty"`
+	Time  *time.Time `json:"time,omitempty"`
+}
+
+type GithubTeam struct {
+	ID     string `json:"id,omitempty"`
+	NodeID string `json:"node_id,omitempty"`
+	Name   string `json:"name,omitempty"`
+	Slug   string `json:"slug,omitempty"`
+}
+
+type GithubPullRequestReview struct {
+	ID          string      `json:"id,omitempty"`
+	NodeID      string      `json:"node_id,omitempty"`
+	User        *GithubUser `json:"user,omitempty"`
+	SubmittedAt *time.Time  `json:"submitted_at,omitempty"`
+	// State can have values CHANGES_REQUESTED, APPROVED, DISMISSED, COMMENTED
+	State             string `json:"state,omitempty"`
+	AuthorAssociation string `json:"author_association,omitempty"`
+}
+
+func NewGithubCommit(in *github.RepositoryCommit) (*GithubCommit, error) {
```

Comment:
I don't see any path to return a non-nil error, please remove it from all signatures
```suggestion
func NewGithubCommit(in *github.RepositoryCommit) *GithubCommit {
```